build/scripts/build.ps1

```diff
@@ -1,84 +1,401 @@
     }
 
-    . (Join-Path $PSScriptRoot "build-utils.ps1")
+    Stop-BuildProcesses
+    & ".\Binaries\$buildConfiguration\Exes\Perf.Runner\Roslyn.Test.Performance.Runner.exe"  $extraArgs --search-directory=".\\Binaries\\$buildConfiguration\\Dlls\\" --no-trace-upload
+    if (-not $?) { 
+        throw "Perf run failed"
+    }
+}
+
+function Test-XUnitCoreClr() { 
+
+    $unitDir = Join-Path $binariesDir "CoreClrTest"
+    $logDir = Join-Path $unitDir "xUnitResults"
+    $logFile = Join-Path $logDir "TestResults.xml"
+    Create-Directory $logDir 
+
+    Write-Host "Publishing CoreClr tests"
+    Run-MSBuild "src\Test\DeployCoreClrTestRuntime\DeployCoreClrTestRuntime.csproj /m /v:m /t:Publish /p:RuntimeIdentifier=win7-x64 /p:PublishDir=$unitDir" | Out-Host
+
+    $corerun = Join-Path $unitDir "CoreRun.exe"
+    $args = Join-Path $unitDir "xunit.console.netcore.exe"
+    foreach ($dll in Get-ChildItem -re -in "*.UnitTests.dll" $unitDir) {
+        $args += " $dll";
+    }
+
+    $args += " -parallel all"
+    $args += " -xml $logFile"
+
+    Write-Host "Running CoreClr tests"
+    Exec-Command $corerun $args | Out-Host
+}
+
+# Core function for running our unit / integration tests tests
+function Test-XUnit() { 
+
+    if ($testCoreClr) {
+        Test-XUnitCoreClr
+        return
+    }
+
+    # To help the VS SDK team track down their issues around install via build temporarily 
+    # re-enabling the build based deployment
+    # 
+    # https://github.com/dotnet/roslyn/issues/17456
+    $deployExtensionViaBuild = $false
+
+    if ($build) {
+        $deployArg = if ($deployExtensionViaBuild) { "true" } else { "false" }
+        Run-MSBuild "Roslyn.sln /p:Configuration=$buildConfiguration /p:DeployExtension=$deployArg"
+        
+        if ($testDesktop) { 
+            Run-MSBuild "src\Samples\Samples.sln /p:Configuration=$buildConfiguration /p:DeployExtension=false"
+        }
+
+        Stop-BuildProcesses
+    }
+
+    $anyVsi = $testVsi -or $TestVsiNetCore
+    if ($anyVsi -and (-not $deployExtensionViaBuild)) {
+        Deploy-VsixViaTool
+    }
+
+    $unitDir = Join-Path $configDir "UnitTests"
+    $runTests = Join-Path $configDir "Exes\RunTests\RunTests.exe"
+    $xunitDir = Join-Path (Get-PackageDir "xunit.runner.console") "tools"
+    $args = "$xunitDir"
 
-    $nuget = Ensure-NuGet
-    if ($clearPackageCache) {
-        Clear-PackageCache
+    if ($testDesktop) {
+        if ($test32) {
+            $dlls = Get-ChildItem -re -in "*.UnitTests.dll" $unitDir
+        }
+        else {
+            $dlls = Get-ChildItem -re -in "*.UnitTests.dll" -ex "*Roslyn.Interactive*" $unitDir 
+        }
     }
+    elseif ($testVsi) {
+        $dlls = Get-ChildItem -re -in "*.IntegrationTests.dll" $unitDir
+    }
+    else {
+        $dlls = Get-ChildItem -re -in "*.IntegrationTests.dll" $unitDir
+        $args += " -trait:Feature=NetCore"
+    }
+
+    if ($cibuild) {
+        $args += " -xml -timeout:50"
```

Comment:
Could you add a comment indicating the time unit? I have no clue if this is 50ms, 50s, 50m, etc